src/boat/helpers.ts

```diff
@@ -0,0 +1,221 @@
+ * @return {boolean}
+ */
+export const isBoatHorizontal = (direction: BoatDirection): boolean => {
+  return direction === BoatDirection.WEST || direction === BoatDirection.EAST;
+};
+
+/**
+ * Check if boat has x axis reversed.
+ *
+ * @param {BoatDirection} direction Boat direction
+ * @return {boolean}
+ */
+export const hasXAxisReversed = (direction: BoatDirection): boolean => {
+  return direction === BoatDirection.EAST || direction === BoatDirection.SOUTH;
+};
+
+/**
+ * Check if boat has y axis reversed.
+ *
+ * @param {BoatDirection} direction Boat direction
+ * @return {boolean}
+ */
+export const hasYAxisReversed = (direction: BoatDirection): boolean => {
+  return direction === BoatDirection.WEST || direction === BoatDirection.SOUTH;
+};
+
+/**
+ * Parse a boat.
+ *
+ * @param {any} boat Boat
+ * @return {Boat}
+ */
+export const parseBoat = (boat: any): Boat => ({
+  lengthCell: boat?.lengthCell || 1,
+  name: boat?.name || '',
+  src: boat?.src || '',
+  widthCell: boat?.widthCell || 1,
+});
+
+/**
+ * Parse boats.
+ *
+ * @param {any} boats Boats
+ * @return {Boat[]}
+ */
+export const parseBoats = (boats: any): Boat[] =>
+  ArrayHelpers.isArray(boats) ? boats.map((boat: any) => parseBoat(boat)) : [];
+
+/**
+ * Parse authorized boat.
+ *
+ * @param {any} authorizedBoat Authorized boat
+ * @return {AuthorizedBoat}
+ */
+export const parseAuthorizedBoat = (authorizedBoat: any): AuthorizedBoat => ({
+  authorisedNumber: authorizedBoat?.authorisedNumber || 0,
+  boat: parseBoat(authorizedBoat?.boat),
+});
+
+/**
+ * Parse authorized fleet.
+ *
+ * @param {any} authorisedFleet Authorized fleet
+ * @return {Boat[]}
+ */
+export const parseAuthorizedFleet = (authorisedFleet: any): Boat[] => {
+  // FIXME: change this bad parsed
+  const newAuthorisedFleet = ArrayHelpers.isArray(authorisedFleet) ? authorisedFleet : [];
+  const boats: Boat[] = [];
+
+  for (const authorisedFleetItem of newAuthorisedFleet) {
+    const authorizedBoat = parseAuthorizedBoat(authorisedFleetItem);
+
+    for (let index = 0; index < authorizedBoat.authorisedNumber; index++) {
+      boats.push(authorizedBoat.boat);
+    }
+  }
+
+  return ArrayHelpers.isEmpty(boats) ? parseBoats(newAuthorisedFleet) : boats;
+};
+
+/**
+ * Get boat bow cells.
+ *
+ * @param {BoardBoat} boardBoat Board boat.
+ * @return {ApiPosition[]}
+ */
+export const getBoatBowCells = (boardBoat: BoardBoat): ApiPosition[] => {
+  const bowCells: ApiPosition[] = [];
+
+  for (let index = 0; index < boardBoat.widthCell; index++) {
+    if (isBoatHorizontal(boardBoat.direction)) {
+      if (hasYAxisReversed(boardBoat.direction)) {
+        bowCells.push([boardBoat.x, boardBoat.y - index]);
+      } else {
+        bowCells.push([boardBoat.x, boardBoat.y + index]);
+      }
+    } else {
+      if (hasXAxisReversed(boardBoat.direction)) {
+        bowCells.push([boardBoat.x - index, boardBoat.y]);
+      } else {
+        bowCells.push([boardBoat.x + index, boardBoat.y]);
+      }
+    }
```

Comment:
With early return, this part should looks like this (rewrite it in your IDE) 

```suggestion
    if (isBoatHorizontal(boardBoat.direction)) {
    
      if (hasYAxisReversed(boardBoat.direction)) {
        bowCells.push([boardBoat.x, boardBoat.y - index]);
        return;
      } 
      
        bowCells.push([boardBoat.x, boardBoat.y + index]);    
      return;
    } 
    

      if (hasXAxisReversed(boardBoat.direction)) {
        bowCells.push([boardBoat.x - index, boardBoat.y]);
        return;
      } 
      
        bowCells.push([boardBoat.x + index, boardBoat.y]);
      
   
```

Reply:
Je ne peux pas faire des retours en pleins dans ma fonction car je boucle justement pour mettre les positions dans `bowCells` et après je retourne `bowCells`. Là en l'occurrence ça renverrai `undefined` alors qu'on s'attends à un retour de type `ApiPosition[]`

Reply:
C'est sur que tu peux le faire sans else, j'approve quand même